devito/archinfo.py

```diff
@@ -114,8 +118,16 @@ def get_cpu_brand():
             except KeyError:
                 warning("Physical core count autodetection failed")
                 physical = 1
-    cpu_info['physical'] = physical
+    try:
+        if physical == (lscpu()['Core(s) per socket'] * lscpu()['Socket(s)']):
```

Comment:
1) the lines above also access `lscpu`, so it's time to assign `out = lscpu()`
2) now we're **always** calling `lscpu()` on startup, which has a cost... how long does it take? order of milliseconds I presume. Can you time it?
2) the  way you wrote this `if` is a bit ugly, isn't it better as

```
expected = out[...]*out[...]
if physical != expected:
    physical = expected
```

note that I dropped the `warning` which would only confuse the user. However, there's a bigger conceptual flaw... :

I would guess the check starting at line 94 (ie the body of the `if lines`) produces a **wrong** `physical` value on some ARM architectures, right? if so, why is that the case? you may say a bug in `psutil`. Fine, but I think we should rather change the code such that the `if` at line 110 **is entered** (this's the fallback branch after all), thus avoiding the need for 121-...

You're instead _adding_ new code (lines 121...) which will be executed _regardless_ of the content of physical, and this may slow down startup, which instead should be extremely quick.

IOW:

* lines 94-108 is where we should sanity-check the content of `lines` such that we're sure it will produce the right `physical` value. 
* lines 110-120 is where we use the fallback, which includes calling `lscpu`

Does this make sense?